### DISCUSSION TITLE
Inverse pole figure for average von Mises stresses in grains

### DISCUSSION THREAD

**Discussion by Shuv20** [mtex-toolbox/mtex#655] [Ask Anything]
Hi all,

I have a question about MTEX and would be grateful for any suggestions!

I have data for average von Mises stresses for grains in FCC polycrystals with n=5, 10. Now, I would like to present them in pole figures or inverse pole figures.  

One thing I tried is that I have drawn two pole figures using MTEX by assuming the grain weights to be proportional to the average von 
Mises stresses, for both n=5, 10. This shows the variations of the average von Mises stresses between grains; however, the means for the two pole figures are identical  (as I assumed the grain weights to be proportional to the average von Mises stresses). However, what I would like to do is this:
Can we plot two inverse pole figures on one scale such that the color range for each of the pole figures represents the mean as well as variations for the average von Mises stresses in the grains? That is the means, together with the variations, for n=5 and 10 correspond to appropriate color in a single colorbar for two inverse pole figures.

Thanks in advance for your help,

**Comment by ralfHielscher** [comment]
Hi Shuvrangsu,

I have some troubles understanding what you want to do. If you just want to adapt the color mapping you might want to have a look at the command `setColorRange` https://mtex-toolbox.github.io/setColorRange.html
Otherwise, it might help to attach some code and figures to illustrate what you want to do.

All the best,

Ralf.

**Reply by Shuv20** [reply]
Hi Ralf,
Thanks for your reply. Please let me elaborate on what I want to do. 

The attached figure shows the standard deviations of the equivalent stress and strain rate in grains for HCP  polycrystals in a spherical triangle representing the fundamental region, 
![SD_IPF](https://user-images.githubusercontent.com/76626057/103212121-eb83a400-4901-11eb-923f-adc734f66554.PNG)
Note that the color range is identical for all the figures, characterizing the mean as well as variations. For example, in Fig 4a, the values for SD(\sigma_e) are around 0.2 while the variations between grains are relatively weak. On the other hand,  in Fig 4e, the mean of SD is 1.1, with significant variations between grains in this case.  The associated color range shows all this information. 

Now, I can do this kind of plot for HCP polycrystals since the fundamental region is defined easily. However, I am having trouble with FCC polycrystals, for which the fundamental region definition is a bit complicated.  Hence, my question is:

How can I plot the standard deviations (SD) of the equivalent stress (or strain-rate) in grains in a spherical triangle representing the fundamental region of FCC polycrystals?  I would like to plot the SDs for FCC polycrystals with nonlinearity exponents n=5,10, and the plots will show the differences in the values as well as variations for n=5 and 10 in a single color range (like the attached figure).

Please let me know  your suggestions.

**Comment by ralfHielscher** [comment]
Hi Shuvrangsu,

if I understand you correctly your central question is how to plot a function (in your case the SD of stress and strain) into the fundamental triangle of the symmetry group.

Here is one approach

1. put the computation of the SD(stress) into a function, e.g. `SDStress(v)` that takes a direction and computes the corresponding standard deviation.
2. define a MTEX spherical function as `fun = S2FunHandle( @(v) SDStress(v) )`
3. define crystal symmetry: `cs=crystalSymmetry('432')`
3. plot the function into the fundamental sector: `plot(fun,cs.fundamentalSector)`

It depends a bit on how you compute the standard deviation of the stress.

Ralf.

**Reply by Shuv20** [reply]
Hi Ralf,
Thanks for your reply. 

The function  **fun = S2FunHandle( @(v) SDStress(v) )** seems to require **vector3d**  v. However, what I have is the Euler angles (in Bunge convention) for grains  and the standard deviations of the equivalent stresses SD(\sigma_e) in those grains. To elaborate, the data set I have:
**Euler1 Euler2 Euler3 SD(\sigma_e)** 
for discrete orientations of grains in full orientational space. 
Is there any way to plot this set of data in the fundamental triangle of the symmetry group?

Happy new year!

**Reply by ralfHielscher** [reply]
What is the connection between the Euler angles and the fundamental triangle of the symmetry group?
